src/mw-utils/services/mw_loading_service.js

```diff
@@ -0,0 +1,142 @@
+'use strict';
+
+angular.module('mwUI.Utils')
+
+/**
+ * @ngdoc service
+ * @name mwUI.Utils.service:Loading
+ *
+ * @description
+ * This service manages loading processes. It can be used e.g. by a httpInterceptor to register a loading process
+ */
+  .service('Loading', function ($timeout) {
+
+    var itemsToLoad = 0,
+      itemsAlreadyLoaded = 0,
+      loading = false,
+      keysToLoad = {},
+      waitUntilDoneTimout,
+      doneCallbacks = [],
+      startCallbacks = [];
+
+    var reset = function () {
+      itemsToLoad = 0;
+      itemsAlreadyLoaded = 0;
+    };
+
+    var executeCallbacks = function (cbs, args, scope) {
```

Comment:
Didn't know that the letters a,l,l,a,c,k are expensive. Please fix param names for better reading. Also args should be arguments